### ISSUE TITLE
npm run dev thrown Unknown error from PostCSS plugin on a vanilla Nextjs app 

### ISSUE THREAD

**Issue by thebreaktop** [vercel/next.js#24942]
### What version of Next.js are you using?

10.2.0

### What version of Node.js are you using?

14.16.1

### What browser are you using?

google Chrom Version 90.0.4430.93

### What operating system are you using?

Mac OS Catalina 10.15.7

### How are you deploying your application?

npm run dev

### Describe the Bug

-  I created a brand new next js app using the command `npx create-next-app`
- I then changed director to where the app is located
- I then executed `npm run dev`
- I see the following error:

>     > my-app@0.1.0 dev /Users/dev/my-app
>     > next dev
>     
>     ready - started server on 0.0.0.0:3000, url: http://localhost:3000
>     info  - Using webpack 5. Reason: no next.config.js https://nextjs.org/docs/messages/webpack5
>     Unknown error from PostCSS plugin. Your current PostCSS version is 8.2.13, but postcss-preset-env uses 7.0.32. Perhaps this is the source of the error below.
>     error - ./node_modules/next/dist/compiled/css-loader/cjs.js??ruleSet[1].rules[2].oneOf[6].use[1]!./node_modules/next/dist/compiled/postcss-loader/cjs.js??ruleSet[1].rules[2].oneOf[6].use[2]!./styles/globals.css
>     TypeError: Cannot read property 'prefix_exceptions' of undefined
>         at Array.map (<anonymous>)

- When I go to `http://localhost:3000/` I see the following error in the terminal

> Error: Cannot find module '/Users/dev/my-app/.next/build-manifest.json'
> Require stack:
> - /Users/dev/my-app/node_modules/next/dist/next-server/server/load-components.js
> - /Users/dev/my-app/node_modules/next/dist/next-server/server/api-utils.js
> - /Users/dev/my-app/node_modules/next/dist/next-server/server/next-server.js
> - /Users/dev/my-app/node_modules/next/dist/server/next.js
> - /Users/dev/my-app/node_modules/next/dist/server/lib/start-server.js
> - /Users/dev/my-app/node_modules/next/dist/cli/next-dev.js
> - /Users/dev/my-app/node_modules/next/dist/bin/next
>     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
>     at Function.mod._resolveFilename (/Users/dev/my-app/node_modules/next/dist/build/webpack/require-hook.js:4:1784)
>     at Function.Module._load (internal/modules/cjs/loader.js:725:27)
>     at Module.require (internal/modules/cjs/loader.js:952:19)
>     at require (internal/modules/cjs/helpers.js:88:18)
>     at loadComponents (/Users/dev/my-app/node_modules/next/dist/next-server/server/load-components.js:1:1691)
>     at async DevServer.findPageComponents (/Users/dev/my-app/node_modules/next/dist/next-server/server/next-server.js:76:257)
>     at async DevServer.renderErrorToHTML (/Users/dev/my-app/node_modules/next/dist/next-server/server/next-server.js:138:266)
>     at async DevServer.render (/Users/dev/my-app/node_modules/next/dist/next-server/server/next-server.js:74:255)
>     at async Object.fn (/Users/dev/my-app/node_modules/next/dist/next-server/server/next-server.js:58:672)
>     at async Router.execute (/Users/dev/my-app/node_modules/next/dist/next-server/server/router.js:25:67)
>     at async DevServer.run (/Users/dev/my-app/node_modules/next/dist/next-server/server/next-server.js:68:1042)
>     at async DevServer.handleRequest (/Users/dev/my-app/node_modules/next/dist/next-server/server/next-server.js:32:504) {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: [
>     '/Users/dev/my-app/node_modules/next/dist/next-server/server/load-components.js',
>     '/Users/dev/my-app/node_modules/next/dist/next-server/server/api-utils.js',
>     '/Users/dev/my-app/node_modules/next/dist/next-server/server/next-server.js',
>     '/Users/dev/my-app/node_modules/next/dist/server/next.js',
>     '/Users/dev/my-app/node_modules/next/dist/server/lib/start-server.js',
>     '/Users/dev/my-app/node_modules/next/dist/cli/next-dev.js',
>     '/Users/dev/my-app/node_modules/next/dist/bin/next'
>   ]
> }

### Expected Behavior

This is a brand new nextjs app. I have not made any changes. After running `npm run dev` the app should start.

### To Reproduce

1. create a new nextjs app e.g. npx create-next-app
2. got to nextjs app directory
3. run npm run dev

I've done the following and still facing the same issue:
- Ive tried `npm cache clean --force ` to clean the cache. I still face the same issues as described above.
- I've tried with the latest node version v16.1.0 and still facing the same issues as described above

**Comment by philliplakis** [comment]
+1

**Comment by noyanyan** [comment]
having same Error message. I'm using Yarn 1.22.10 + Ubuntu 20.04.2 LTS (via wsl2) + Node 15.10.0. 
Even with Node 14.16.0 same message appears.

**Comment by sammkj** [comment]
I'm facing similiar issue as well.

```
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Using webpack 5. Reason: no custom webpack configuration in next.config.js https://nextjs.org/docs/messages/webpack5
error - ../node_modules/next/dist/compiled/css-loader/cjs.js??ruleSet[1].rules[2].oneOf[5].use[1]!../node_modules/next/dist/compiled/postcss-loader/cjs.js??ruleSet[1].rules[2].oneOf[5].use[2]!../node_modules/tailwindcss/tailwind.css
TypeError: Cannot read property 'prefix_exceptions' of undefined
    at Array.map (<anonymous>)
    at Array.map (<anonymous>)
```

**Comment by rickchristie** [comment]
Here's the workaround I did:
* I copied `dependencies` and `devDependencies` in `package.json`, along with the _entire_ `package-lock.json` of a previous next project that was working.
* I removed the entire `node_modules` directory.
* I run `npm install`.

After running the steps above, I can run `npm run dev` without this error appearing on the new project.

**Comment by LucaNerlich** [comment]
There are a lot of post css issues right now, half ish working resolution in this thread, similar to my experience here: https://github.com/vercel/next.js/issues/24696

**Comment by timneutkens** [comment]
```js
create a new nextjs app e.g. npx create-next-app
got to nextjs app directory
run npm run dev
```

These steps do not reproduce the issue, please provide a full reproduction so that it can be investigated.

**Comment by noyanyan** [comment]
Thanks for checking this.  I can't reproduce it or provide repo (I deleted the folder in question), but I think this is a distribution issue.


```
// command in my case(now unable to reproduce). 
yarn create next-app my-app
cd my-app
yarn dev
```

From the fact that comments are concentrated at certain times and no comment after that, I guess the cache or something on the delivery side  got corrupted. That may not be a problem with Next.js.
It is now working fine. Thank you for making Next.js such a great product.

**Comment by sammkj** [comment]
The error went away after I switched from NPM to Yarn. I think @noyanyan might be right that there may be some dependency resolution issue.

**Comment by tkesgar** [comment]
**TL;DR:** I think the issue here is that the installed `postcss` directly in `node_modules` is v7, while the plugins need v8. A quick fix would be by explicitly install PostCSS v8 in your dependency:

```json
{
  "devDependencies": {
    "postcss": "^8"
  }
}
```

---

This is the result of `npm explain` in my repo, which has the issue:

```
$ npm explain postcss
postcss@8.2.13
node_modules/next/node_modules/postcss
  postcss@"8.2.13" from next@10.2.2
  node_modules/next
    next@"^10.2.2" from the root project

postcss@7.0.35
node_modules/postcss
  postcss@"^7.0.32" from autoprefixer@9.8.6
  node_modules/autoprefixer
    autoprefixer@"^9.6.1" from postcss-preset-env@6.7.0
    node_modules/postcss-preset-env
      postcss-preset-env@"^6.7.0" from the root project
    autoprefixer@"^9.8.6" from stylelint@13.13.1
    node_modules/stylelint
      dev stylelint@"^13.13.1" from the root project
  postcss@"^7.0.5" from css-blank-pseudo@0.1.4
<some other lines>
```

The installed `postcss` package directly in `node_modules/postcss` is v7.

When `next build` runs PostCSS, I am guessing that it uses the copy in cwd (our repo root) instead of `next`'s own copy. The result is that PostCSS v7 is loaded, and the build will fail if there are v8-only plugins.

Explicitly installing `postcss@^8` solves the issue, since it now has PostCSS v8 installed at `node_modules/postcss`:

```
$ npm explain postcss
postcss@8.2.15
node_modules/postcss
  postcss@"^8.2.15" from the root project
```

I don't know why using Yarn solves the issue, but I am guessing that it uses a different dependency resolving strategy that coincidentally puts PostCSS 8.

I think a proper fix here would be to add `postcss@^8` as `peerDependencies` of `next`, so that we have to explicitly install `postcss` along with `next`:

```
$ npm install next postcss@^8
```

Another proper fix is to somehow ensure that `import "postcss"`/`require("postcss")` is done from Next.js module cwd (`node_modules/next`), but I am not sure how to do it.

**Comment by pcjmfranken** [comment]
@tkesgar Is the issue really resolved by using Yarn, or are you just no longer seeing the errors? Are `preset-env`'s [polyfills](https://preset-env.cssdb.org/features#stage-3) actually being applied when using Yarn?

**Comment by tkesgar** [comment]
The fix I wrote above is to install postcss v8 explicitly as part of the repo dependency, so that it node_modules/postcss will be v8.

I haven't tried with yarn because my current project uses npm. yarn might have a different module resolution strategy that coincidentally puts postcss v8 in the right place. It can be made explicit by installing the package (yarn add postcss@^8).

**Comment by fredlb** [comment]
I had this issue as well, same ordeal with vanilla next app via create-next-app.
However this only happened when I did this in a nested directory of an already existing JS project:
```
-root/
-- node_modules/
-- my-next-app/ <-- created by create-next-app
---- node_modules/
---- pages/
---- package.json
-- package.json
```
This seems to happen regardless of node and npm version. So maybe there is some strange issue where things are trying to resolve outside `my-next-app`.

**Comment by timneutkens** [comment]
Next.js 12 uses postcss 8 everywhere, so this is probably fixed 👍

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
